woocommerce/woocommerce-gutenberg-products-block
# Add Inner blocks to order summary

The WIP prefix is a leftover, the work on this branch was completed.

This PR converts the Cart Block and Checkout Block "order summary" areas into an inner block structure. 

Cart Block:

- Heading `woocommerce/cart-order-summary-heading`
- Subtotal `woocommerce/cart-order-summary-subtotal`
- Fee `woocommerce/cart-order-summary-fee`
- Shipping `woocommerce/cart-order-summary-shipping`
- Discount `woocommerce/cart-order-summary-discount`
- Coupon Form `woocommerce/cart-order-summary-coupon-form`
- Taxes `woocommerce/cart-order-summary-taxes`

Checkout Block

- Cart Items `woocommerce/checkout-order-summary-cart-items`
- Subtotal `woocommerce/checkout-order-summary-subtotal`
- Fee `woocommerce/checkout-order-summary-fee`
- Shipping `woocommerce/checkout-order-summary-shipping`
- Discount `woocommerce/checkout-order-summary-discount`
- Coupon Form `woocommerce/checkout-order-summary-coupon-form`
- Taxes `woocommerce/cart-order-summary-taxes`

**NOTE:** Yes there are many similarities between the cart and checkout inner blocks, however, they cannot be shared due to the way we use context providers. If they were shared, both Cart and Checkout Blocks would import the above inner blocks, and those inner blocks would then receive the wrong values from context depending on what was imported first. To work around this we have a unique set of inner blocks per each parent block.

The new inner blocks implement existing slotfills, making this change backwards compatible. Subscriptions extension can be installed to verify this since it extends the order totals area including its own shipping and total row.

Fixes #6059

### Screenshots

#### Cart block before

**The Order summary block**

<img width="747" alt="Screenshot 2022-03-29 at 15 47 30" src="https://user-images.githubusercontent.com/1628454/160852502-888753fe-eb6a-4cba-81af-e8e12d042e71.png">

#### Cart block after

**The Order summary block**
<img width="610" alt="Screenshot 2022-03-29 at 15 48 20" src="https://user-images.githubusercontent.com/1628454/160852500-51337fac-59ad-48f7-8bf0-642c75c61c7f.png">

**Selection of an Order summary Inner block**
<img width="611" alt="Screenshot 2022-03-29 at 15 48 32" src="https://user-images.githubusercontent.com/1628454/160852488-30f8277c-ebd3-4edc-86a3-00bcd451b71d.png">







### Testing

#### Automated Tests

Existing block tests cover these inner blocks since the functionality has just moved down a level.

  
### Manual Testing

How to test the changes in this Pull Request:

1. Go to a page and add the Checkout block.
2. Go to a page and add the Cart block
3. In the editor, inside each block, select the Order summary and notice that you can select the following inner blocks:

| Cart | Checkout |
| --------------- | --------------- | 
|  Heading, Subtotal, Cart Items, Fees, Coupon Form, Shipping, Taxes | Subtotal, Cart Items, Fees, Coupon Form, Shipping, Taxes | 
4. Try to move some of the inner blocks and save. Confirm that in the Cart block only Order Summary Heading and the Coupon form inner blocks can be removed, and for the Checkout block only the Coupon form.
5. Check on the website that your changes are reflected. Note that Taxes, Fees and Coupons will appear on the website only if the shop supports them.
6. For the Cart > Shipping inner block make sure you can enable disable the shipping calculator and that the change is visibile in the website
7. Enable Taxes on your website (`/wp-admin/admin.php?page=wc-settings`) by clicking  `Enable tax rates and calculations`
8. Go to the C & C blocks and select the Taxes inner block. 
9. In the inner block's setting toggle on / of `Show rate after tax name` and make sure that is reflected in the website
10. Try to place an order and make sure it's successful.




### User Facing Testing

These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).


### Changelog

> The order summary area for the Cart and Checkout Blocks is now powered by Inner Blocks allowing for more customizations and extensibility.
